lib/repl.js

```diff
@@ -1455,14 +1455,17 @@ function isRecoverableError(e, code) {
   if (e && e.name === 'SyntaxError') {
     var message = e.message;
     if (message === 'Unterminated template literal' ||
-        message === 'Missing } in template expression') {
+        message === 'Unexpected end of input') {
       return true;
     }
 
-    if (message.startsWith('Unexpected end of input') ||
-        message.startsWith('missing ) after argument list') ||
-        message.startsWith('Unexpected token'))
-      return true;
+    if (message === 'missing ) after argument list') {
+      const frames = e.stack.split(/\r?\n/);
+      const pos = frames.findIndex((f) => f.match(/^\s*\^+$/));
+      if (pos > 0 && frames[pos - 1].length === frames[pos].length)
+        return true;
+      else return false;
```

Comment:
Nit: please remove the else. This is a style that is normally not used here.

Reply:
rebasing and amending to same commit